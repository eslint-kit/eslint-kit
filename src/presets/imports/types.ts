export interface Options {
  sort?: {
    newline?: boolean
    groups?: string[][]
  }
  alias?: {
    root?: string
    paths?: Record<string, string>
    jsconfig?: string
  }
  layout?: boolean
}
