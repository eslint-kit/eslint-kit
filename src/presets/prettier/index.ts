import { Config } from 'prettier'
import { publicPresetNames } from '../names'
import { createPreset } from '../shared'

export const prettier = createPreset<Config>({
  name: publicPresetNames.prettier,
  compile: ({ options }) => ({
    plugins: ['prettier'],
    rules: {
      'prettier/prettier': options ? ['warn', options] : 'warn',
    },
  }),
})
