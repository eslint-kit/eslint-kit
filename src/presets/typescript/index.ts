import path from 'path'
import { Linter } from '@typescript-eslint/utils/dist/ts-eslint'
import { EXTENSIONS, importExtensionsRule } from '../../shared'
import { createPreset } from '../shared'

export interface Options {
  root?: string
  tsconfig?: string
}

export const typescriptRules: Linter.RulesRecord = {
  '@typescript-eslint/adjacent-overload-signatures': 'warn',
  '@typescript-eslint/ban-ts-comment': 'error',
  '@typescript-eslint/ban-types': 'error',
  '@typescript-eslint/naming-convention': [
    'warn',
    {
      selector: 'variable',
      format: ['camelCase', 'UPPER_CASE', 'PascalCase'],
      leadingUnderscore: 'allow',
    },
    {
      selector: 'parameter',
      format: ['camelCase'],
      leadingUnderscore: 'allow',
    },
    {
      selector: 'typeLike',
      format: ['PascalCase'],
    },
  ],
  '@typescript-eslint/consistent-type-assertions': 'warn',
  '@typescript-eslint/consistent-type-definitions': ['warn', 'interface'],
  '@typescript-eslint/no-array-constructor': 'warn',
  '@typescript-eslint/no-empty-interface': 'warn',
  '@typescript-eslint/no-explicit-any': 'warn',
  '@typescript-eslint/no-extra-non-null-assertion': 'warn',
  '@typescript-eslint/no-for-in-array': 'warn',
  '@typescript-eslint/no-inferrable-types': 'error',
  '@typescript-eslint/no-misused-new': 'error',
  '@typescript-eslint/no-namespace': 'error',
  '@typescript-eslint/no-non-null-assertion': 'warn',
  '@typescript-eslint/no-unused-vars': [
    'warn',
    {
      ignoreRestSiblings: true,
      varsIgnorePattern: '^_',
      argsIgnorePattern: '^_',
    },
  ],
  '@typescript-eslint/no-unnecessary-condition': 'warn',
  '@typescript-eslint/no-unused-expressions': 'warn',
  '@typescript-eslint/no-useless-constructor': 'warn',
  '@typescript-eslint/prefer-for-of': 'warn',
  '@typescript-eslint/prefer-includes': 'warn',
  '@typescript-eslint/prefer-namespace-keyword': 'error',
  '@typescript-eslint/prefer-optional-chain': 'warn',
  '@typescript-eslint/prefer-string-starts-ends-with': 'warn',
  '@typescript-eslint/require-array-sort-compare': 'error',
  '@typescript-eslint/return-await': ['error', 'in-try-catch'],
  '@typescript-eslint/triple-slash-reference': 'error',

  // Specific rules that are enabled using @typescript-eslint, but have analogues in common eslint
  'camelcase': 'off',
  'no-array-constructor': 'off',
  'no-unused-vars': 'off',
  'no-unused-expressions': 'off',
  'no-useless-constructor': 'off',
  'no-return-await': 'off',

  // Checked by Typescript - ts(2378)
  'getter-return': 'off',
  // Checked by Typescript - ts(2300)
  'no-dupe-args': 'off',
  // Checked by Typescript - ts(1117)
  'no-dupe-keys': 'off',
  // Checked by Typescript - ts(7027)
  'no-unreachable': 'off',
  // Checked by Typescript - ts(2367)
  'valid-typeof': 'off',
  // Checked by Typescript - ts(2588)
  'no-const-assign': 'off',
  // Checked by Typescript - ts(2588)
  'no-new-symbol': 'off',
  // Checked by Typescript - ts(2376)
  'no-this-before-super': 'off',
  // This is checked by Typescript using the option `strictNullChecks`.
  'no-undef': 'off',
  // This is already checked by Typescript.
  'no-dupe-class-members': 'off',
  // This is already checked by Typescript.
  'no-redeclare': 'off',

  // Checked by Typescript
  'import/no-extraneous-dependencies': 'off',
}

export const typescript = createPreset<'typescript', Options | void>({
  name: 'typescript',
  updateMeta: ({ meta, options }) => {
    meta.extensions = meta.extensions.concat(EXTENSIONS.TS)
    meta.typescript.used = true
    meta.typescript.root = path.resolve(meta.root, options?.root ?? './')
    meta.typescript.tsconfig = options?.tsconfig ?? 'tsconfig.json'
  },
  compile: ({ meta }) => ({
    plugins: ['import', '@typescript-eslint'],
    parser: '@typescript-eslint/parser',
    parserOptions: {
      project: meta.typescript.tsconfig,
      createDefaultProgram: true,
      tsconfigRootDir: meta.typescript.root,
    },
    settings: {
      'import/extensions': meta.extensions,
      'import/resolver': {
        node: {
          extensions: meta.extensions,
        },
      },
      'import/ignore': ['\\.(coffee|scss|css|less|hbs|svg|json)$'],
      'import/external-module-folders': ['node_modules', 'node_modules/@types'],
      'import/parsers': {
        '@typescript-eslint/parser': EXTENSIONS.TS,
      },
    },
    rules: {
      'import/extensions': [
        importExtensionsRule[0],
        importExtensionsRule[1],
        meta.extensions.reduce<{ [key: string]: 'never' }>((acc, extension) => {
          const extensionShort = extension.slice(1)
          acc[extensionShort] = 'never'
          return acc
        }, {}),
      ],
    },
    overrides: [
      {
        files: ['*.ts', '*.tsx'],
        rules: typescriptRules,
      },
    ],
  }),
})
