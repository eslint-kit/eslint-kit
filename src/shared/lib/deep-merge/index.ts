/* eslint-disable @typescript-eslint/no-explicit-any */
import { Merge, PathsOf } from './types'

function isObject(value: any): value is Record<string, any> {
  return typeof value === 'object' && value !== null
}

export enum Strategy {
  Shallow,
  Deep,
  Override,
}

type AllPaths<A, B> = PathsOf<A> | PathsOf<B> | null

function shallowMerge<A, B>(a: A, b: B): Merge<A, B> {
  let merged: any

  if (Array.isArray(a) && Array.isArray(b)) {
    merged = [...a, ...b]
  } else if (isObject(a) && isObject(b)) {
    merged = { ...a, ...b }
  } else if (typeof b !== 'undefined') {
    merged = b
  } else {
    merged = a
  }

  return merged as Merge<A, B>
}

// eslint-disable-next-line sonarjs/cognitive-complexity
export function deepMerge<A, B>(
  a: A,
  b: B,
  strategy: (path: AllPaths<A, B>) => Strategy = () => Strategy.Deep,
  basePath: AllPaths<A, B> = null,
): Merge<A, B> {
  if (strategy(basePath) === Strategy.Shallow) {
    return shallowMerge(a, b)
  }

  if (strategy(basePath) === Strategy.Override) {
    return b as Merge<A, B>
  }

  if (Array.isArray(a) && Array.isArray(b)) {
    return shallowMerge(a, b)
  }

  if (isObject(a) && isObject(b)) {
    const keys = [...Object.keys(a), ...Object.keys(b)]
    const final: any = {}

    for (const key of keys) {
      const path = basePath ? [basePath, key].join('.') : key
      final[key] = deepMerge(
        (a as any)[key],
        (b as any)[key],
        strategy,
        path as AllPaths<A, B>,
      )
    }

    return final
  }

  return shallowMerge(a, b)
}
