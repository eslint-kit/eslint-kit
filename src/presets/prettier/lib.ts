import { Linter } from 'eslint'
import { Options } from 'prettier'
import { Meta } from '../shared'

export interface Input {
  options?: Options
  meta: Meta
}

export function createPrettierRule({ meta, options }: Input) {
  const plugins: string[] = []

  if (meta.imports.layout) {
    plugins.push('prettier-plugin-layout')
  }

  return [
    'warn',
    {
      plugins,
      ...options,
    },
  ] as Linter.RuleEntry
}
