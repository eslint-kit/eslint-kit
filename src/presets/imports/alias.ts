import path from 'path'
import { conditional } from '../../shared/lib/eslint'
import { readJson } from '../../shared/lib/fs'
import { Jsconfig } from '../../shared/types'
import { publicPresetNames } from '../names'
import { Meta } from '../shared'
import { Options } from './types'

export type Input = {
  options?: Options
  meta: Meta
}

export function createAliasSettings({ options = {}, meta }: Input) {
  const { alias: aliasOptions = {} } = options
  const {
    root = meta.root,
    paths = {},
    jsconfig = 'jsconfig.json',
  } = aliasOptions

  const jsconfigJson = readJson<Jsconfig>(meta.root, jsconfig)

  const useTsconfig = meta.presets.has(publicPresetNames.typescript)
  const useJsconfig = Boolean(jsconfigJson)

  const alias = Object.entries(paths).reduce<Record<string, string>>(
    (alias, [key, value]) => {
      alias[key] = path.join(root, value)
      return alias
    },
    {},
  )

  return {
    'import-x/resolver': {
      'eslint-import-resolver-custom-alias': {
        alias,
        extensions: meta.imports.extensions,
      },
      ...conditional.settings(useJsconfig, {
        jsconfig: {
          config: path.resolve(meta.root, jsconfig),
          extensions: meta.imports.extensions,
        },
      }),
      ...conditional.settings(useTsconfig, {
        typescript: {
          alwaysTryTypes: true,
          project: meta.typescript.tsconfig,
        },
      }),
    },
  }
}
