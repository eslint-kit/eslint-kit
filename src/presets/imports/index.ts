import { EXTENSIONS } from '../../shared'
import { conditional } from '../../shared/lib/eslint'
import { publicPresetNames } from '../names'
import { createPreset } from '../shared'
import { createAliasSettings } from './alias'
import { createExtensionsRule } from './lib'
import { Options } from './types'

const DEFAULT_IMPORT_GROUPS: string[][] = [
  // side effects
  ['^\\u0000'],

  // node.js libraries and scoped libraries
  ['^(child_process|crypto|events|fs|http|https|os|path)(/.*)?$', '^@?\\w'],

  // common aliases (@app, @root, @/, ~/) and anything not matched
  ['^@app', '^@root', '^~', '^'],

  // relative imports
  ['^\\.'],
]

export const imports = createPreset<Options>({
  name: publicPresetNames.imports,
  updateMeta: ({ meta, options }) => {
    meta.imports.extensions.push(...EXTENSIONS.MISC)
    meta.imports.extensions.push(...EXTENSIONS.JS)

    meta.imports.layout = options?.layout ?? false
  },
  compile: ({ options = {}, meta }) => {
    const { sort = {} } = options
    const { newline = false, groups = DEFAULT_IMPORT_GROUPS } = sort

    const finalGroups = newline ? groups : [groups.flat()]

    return {
      plugins: ['import', 'simple-import-sort'],
      settings: {
        'import/extensions': meta.imports.extensions,
        'import/core-modules': [],
        'import/ignore': ['\\.(coffee|scss|css|less|hbs|svg|json)$'],
        'import/internal-regex': '^@types/',
        ...createAliasSettings({ options, meta }),
      },
      rules: {
        'import/no-unresolved': [
          'error',
          { commonjs: true, caseSensitive: true },
        ],
        'import/no-default-export': 'warn',
        'import/no-anonymous-default-export': 'error',
        'import/export': 'error',
        'import/no-mutable-exports': 'error',
        'import/no-amd': 'error',
        'import/first': 'error',
        'import/no-duplicates': 'error',
        'import/extensions': createExtensionsRule(meta.imports.extensions),
        'import/newline-after-import': 'warn',
        'import/no-webpack-loader-syntax': 'error',
        'import/no-self-import': 'error',
        'import/no-cycle': 'off', // TODO: turn on when execution time became normal
        // 'import/no-cycle': ['error', { maxDepth: Number.POSITIVE_INFINITY }],
        'import/no-useless-path-segments': ['warn', { commonjs: true }],

        'simple-import-sort/imports': ['warn', { groups: finalGroups }],
        'simple-import-sort/exports': 'warn',
      },
      overrides: conditional.overrides(meta.imports.layout, [
        {
          files: EXTENSIONS.JS_AND_TS,
          rules: {
            'import/no-duplicates': 'off',
          },
        },
      ]),
    }
  },
})
