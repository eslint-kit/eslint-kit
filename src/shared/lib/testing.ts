import fs from 'fs'
import path from 'path'
import { ESLint } from 'eslint'
import { Preset, PresetName, PRIORITY, compilePresets } from '../../presets'
import { base } from '../../presets/base'

export interface Options {
  presets: Preset[]
  priority?: PresetName[]
  dirname: string
  files: string[]
  extension: 'js' | 'jsx' | 'ts' | 'tsx' | 'svelte' | 'vue'
}

export async function testConfig({
  presets,
  priority = PRIORITY,
  dirname,
  files,
  extension,
}: Options) {
  const root = path.resolve(dirname, './tests')

  const cli = new ESLint({
    baseConfig: compilePresets([base({ root }), ...presets], priority),
    useEslintrc: false,
    cwd: root,
    ignore: false,
  })

  for (const file of files) {
    const filePath = path.resolve(root, `./${file}.${extension}`)
    const code = fs.readFileSync(filePath).toString()
    const result = await cli.lintText(code, { filePath })
    expect(result).toMatchSnapshot(`${file}`)
  }
}
