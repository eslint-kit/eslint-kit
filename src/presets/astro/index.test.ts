import { testConfig } from '../../shared/lib/testing'
import { imports } from '../imports'
import { prettier } from '../prettier'
import { typescript } from '../typescript'
import { astro } from './index'

describe('[Presets] Astro', () => {
  it('should match snapshots', async () => {
    await testConfig({
      presets: [astro()],
      dirname: __dirname,
      files: ['basic-clear', 'basic-warn', 'basic-error'],
      extension: 'js',
    })
  })

  it('should lint .astro files', async () => {
    await testConfig({
      presets: [astro()],
      dirname: __dirname,
      files: ['astro-clear', 'astro-warn', 'astro-error'],
      extension: 'astro',
    })
  })

  it('should work with typescript', async () => {
    await testConfig({
      presets: [typescript(), astro()],
      dirname: __dirname,
      files: ['basic-clear', 'basic-warn', 'basic-error'],
      extension: 'ts',
    })
  })

  it('should lint typescript in .astro files', async () => {
    await testConfig({
      presets: [typescript(), astro()],
      dirname: __dirname,
      files: ['ts-astro-clear', 'ts-astro-warn', 'ts-astro-error'],
      extension: 'astro',
    })
  })

  it('should not conflict with prettier', async () => {
    await testConfig({
      presets: [prettier(), astro()],
      dirname: __dirname,
      files: ['astro-warn'],
      extension: 'astro',
    })
  })

  it('should not complain about astro imports', async () => {
    await testConfig({
      presets: [imports(), astro()],
      dirname: __dirname,
      files: ['astro-import-clear', 'astro-import-error'],
      extension: 'astro',
    })
  })

  it('should not complain about astro imports (typescript)', async () => {
    await testConfig({
      presets: [typescript(), imports(), astro()],
      dirname: __dirname,
      files: ['ts-astro-import-clear', 'ts-astro-import-error'],
      extension: 'astro',
    })
  })
})
