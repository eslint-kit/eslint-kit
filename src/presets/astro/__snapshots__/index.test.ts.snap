// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Presets] Astro should lint .astro files: astro-clear 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/astro-clear.astro",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should lint .astro files: astro-error 1`] = `
Array [
  Object {
    "errorCount": 1,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/astro-error.astro",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 22,
        "endColumn": 40,
        "endLine": 2,
        "line": 2,
        "message": "'Astro.canonicalURL' is deprecated. Use 'Astro.url' helper instead.",
        "messageId": "deprecated",
        "nodeType": "MemberExpression",
        "ruleId": "astro/no-deprecated-astro-canonicalurl",
        "severity": 2,
      },
    ],
    "source": "---
const canonicalURL = Astro.canonicalURL
console.info(canonicalURL)
---

<html>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
<style>
  h1 {
    color: orange;
  }
</style>
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should lint .astro files: astro-warn 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/astro-warn.astro",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 23,
        "endColumn": 32,
        "endLine": 10,
        "line": 10,
        "message": "'textColor' is defined but never used.",
        "messageId": "unused",
        "nodeType": "Identifier",
        "ruleId": "astro/no-unused-define-vars-in-style",
        "severity": 1,
      },
    ],
    "source": "---
const foregroundColor = 'rgb(221 243 228)'
---

<html>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
<style define:vars={{ textColor: foregroundColor }}>
  h1 {
    color: orange;
  }
</style>
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 1,
  },
]
`;

exports[`[Presets] Astro should lint typescript in .astro files: ts-astro-clear 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/ts-astro-clear.astro",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should lint typescript in .astro files: ts-astro-error 1`] = `
Array [
  Object {
    "errorCount": 1,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/ts-astro-error.astro",
    "fixableErrorCount": 1,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 12,
        "endColumn": 18,
        "endLine": 2,
        "fix": Object {
          "range": Array [
            15,
            21,
          ],
          "text": "number",
        },
        "line": 2,
        "message": "Don't use \`Number\` as a type. Use number instead",
        "messageId": "bannedTypeMessage",
        "nodeType": "Identifier",
        "ruleId": "@typescript-eslint/ban-types",
        "severity": 2,
      },
    ],
    "source": "---
const one: Number = 1
console.info(one)
---

<html>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
<style>
  h1 {
    color: orange;
  }
</style>
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should lint typescript in .astro files: ts-astro-warn 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/ts-astro-warn.astro",
    "fixableErrorCount": 0,
    "fixableWarningCount": 1,
    "messages": Array [
      Object {
        "column": 6,
        "endColumn": 10,
        "endLine": 2,
        "fix": Object {
          "range": Array [
            4,
            16,
          ],
          "text": "interface Test ",
        },
        "line": 2,
        "message": "Use an \`interface\` instead of a \`type\`.",
        "messageId": "interfaceOverType",
        "nodeType": "Identifier",
        "ruleId": "@typescript-eslint/consistent-type-definitions",
        "severity": 1,
      },
    ],
    "source": "---
type Test = {
  one: number
}
const test: Test = { one: 1 }
console.info(test)
---

<html>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
<style>
  h1 {
    color: orange;
  }
</style>
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 1,
  },
]
`;

exports[`[Presets] Astro should match snapshots: basic-clear 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/basic-clear.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should match snapshots: basic-error 1`] = `
Array [
  Object {
    "errorCount": 1,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/basic-error.js",
    "fixableErrorCount": 1,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 1,
        "endColumn": 12,
        "endLine": 1,
        "fix": Object {
          "range": Array [
            0,
            3,
          ],
          "text": "let",
        },
        "line": 1,
        "message": "Unexpected var, use let or const instead.",
        "messageId": "unexpectedVar",
        "nodeType": "VariableDeclaration",
        "ruleId": "no-var",
        "severity": 2,
      },
    ],
    "source": "var one = 1
export { one }
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should match snapshots: basic-warn 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/basic-warn.js",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 7,
        "endColumn": 10,
        "endLine": 1,
        "line": 1,
        "message": "'one' is assigned a value but never used. Allowed unused vars must match /^_/u.",
        "messageId": "unusedVar",
        "nodeType": "Identifier",
        "ruleId": "no-unused-vars",
        "severity": 1,
      },
    ],
    "source": "const one = 1
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 1,
  },
]
`;

exports[`[Presets] Astro should not conflict with prettier: astro-warn 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/astro-warn.astro",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 23,
        "endColumn": 32,
        "endLine": 10,
        "line": 10,
        "message": "'textColor' is defined but never used.",
        "messageId": "unused",
        "nodeType": "Identifier",
        "ruleId": "astro/no-unused-define-vars-in-style",
        "severity": 1,
      },
    ],
    "source": "---
const foregroundColor = 'rgb(221 243 228)'
---

<html>
  <body>
    <h1>Hello, World!</h1>
  </body>
</html>
<style define:vars={{ textColor: foregroundColor }}>
  h1 {
    color: orange;
  }
</style>
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 1,
  },
]
`;

exports[`[Presets] Astro should work with typescript: basic-clear 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/basic-clear.ts",
    "fixableErrorCount": 0,
    "fixableWarningCount": 0,
    "messages": Array [],
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should work with typescript: basic-error 1`] = `
Array [
  Object {
    "errorCount": 1,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/basic-error.ts",
    "fixableErrorCount": 1,
    "fixableWarningCount": 0,
    "messages": Array [
      Object {
        "column": 12,
        "endColumn": 18,
        "endLine": 1,
        "fix": Object {
          "range": Array [
            11,
            17,
          ],
          "text": "number",
        },
        "line": 1,
        "message": "Don't use \`Number\` as a type. Use number instead",
        "messageId": "bannedTypeMessage",
        "nodeType": "Identifier",
        "ruleId": "@typescript-eslint/ban-types",
        "severity": 2,
      },
    ],
    "source": "const one: Number = 1
export { one }
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 0,
  },
]
`;

exports[`[Presets] Astro should work with typescript: basic-warn 1`] = `
Array [
  Object {
    "errorCount": 0,
    "fatalErrorCount": 0,
    "filePath": "<PROJECT_ROOT>/src/presets/astro/tests/basic-warn.ts",
    "fixableErrorCount": 0,
    "fixableWarningCount": 1,
    "messages": Array [
      Object {
        "column": 6,
        "endColumn": 10,
        "endLine": 1,
        "fix": Object {
          "range": Array [
            0,
            12,
          ],
          "text": "interface Test ",
        },
        "line": 1,
        "message": "Use an \`interface\` instead of a \`type\`.",
        "messageId": "interfaceOverType",
        "nodeType": "Identifier",
        "ruleId": "@typescript-eslint/consistent-type-definitions",
        "severity": 1,
      },
      Object {
        "column": 6,
        "endColumn": 10,
        "endLine": 1,
        "line": 1,
        "message": "'Test' is defined but never used. Allowed unused vars must match /^_/u.",
        "messageId": "unusedVar",
        "nodeType": null,
        "ruleId": "@typescript-eslint/no-unused-vars",
        "severity": 1,
      },
    ],
    "source": "type Test = {
  one: number
}
",
    "suppressedMessages": Array [],
    "usedDeprecatedRules": Array [],
    "warningCount": 2,
  },
]
`;
